modules/core/src/lib/attribute/attribute-transition-utils.ts

```diff
@@ -184,7 +188,44 @@ export function padBuffer({
   });
 
   if (buffer.byteLength < target.byteLength + byteOffset) {
-    throw new Error(`Buffer size is immutable, ${buffer.byteLength} bytes`);
+    buffer = buffer.device.createBuffer({byteLength: target.byteLength + byteOffset});
   }
   buffer.write(target, byteOffset);
+  return buffer;
+}
+
+/** @deprecated TODO(v9.1): Buffer reads should be asynchronous and avoid accessing GL context. */
+export function getBufferData(
+  buffer: Buffer,
+  Ctor: TypedArrayConstructor,
+  byteOffset = 0,
+  byteLength = buffer.byteLength
+): TypedArray {
+  const _buffer = buffer as any;
+  _buffer.device.assertWebGL2();
+
+  const dstLength = byteLength / Ctor.BYTES_PER_ELEMENT;
+  const dstArray = new Ctor(dstLength);
+  const dstOffset = 0;
+
+  // Use GL.COPY_READ_BUFFER to avoid disturbing other targets and locking type
+  _buffer.gl.bindBuffer(GL.COPY_READ_BUFFER, _buffer.handle);
+  _buffer.gl2.getBufferSubData(GL.COPY_READ_BUFFER, byteOffset, dstArray, dstOffset, dstLength);
+  _buffer.gl.bindBuffer(GL.COPY_READ_BUFFER, null);
+
+  return dstArray;
+}
+
+export function getAttributeVertexFormat(size: 1 | 2 | 3 | 4): LumaVertexFormat {
```

Comment:
This only supports `float32`, use [getVertexFormatFromAttribute](https://github.com/visgl/luma.gl/blob/master/modules/core/src/adapter/type-utils/vertex-format-from-attribute.ts#L58) instead?

Reply:
It looks like attribute transitions support `float`, `vec2`, `vec3`, and `vec4` types, and to call `getVertexFormatFromAttribute` we'd need to allocate an empty Float32Array... the type of `attribute.value` is actually Float64Array:

https://github.com/visgl/deck.gl/blob/c56c50363ab5e970fc8d4539685d6093aeafb374/modules/core/src/transitions/gpu-interpolation-transition.ts#L120-L122

I've cleaned this up a bit and added JSDoc on the function, but I think it needs to roughly do what it is doing now – let me know if there are other concerns!